cmd/allocator/main.go

```diff
@@ -224,70 +224,38 @@ func main() {
 	h := newServiceHandler(kubeClient, agonesClient, health, conf.MTLSDisabled, conf.TLSDisabled, conf.remoteAllocationTimeout, conf.totalRemoteAllocationTimeout, conf.allocationBatchWaitTime)
 
 	if !h.tlsDisabled {
-		watcherTLS, err := fsnotify.NewWatcher()
+		cancelTLS, err := fswatch.Watch(logger, tlsDir, time.Second, func() {
+			tlsCert, err := readTLSCert()
+			if err != nil {
+				logger.WithError(err).Error("could not load TLS certs; keeping old one")
+			} else {
```

Comment:
nit: Now that this is nicely wrapped in a function, we can use the more idiomatic control flow of

```
if err != nil {
  logger.WithError(...)
  return
}
h.tlsMutex.Lock()
h.tlsCert = tlsCert
h.tlsMutex.Unlock()
logger.Info(...)
```

This also means that if there is an error we won't get an error log line followed by the info line.

Reply:
Another thing that occurred to me (which doesn't make as much difference here) is that instead of defering the Unlock call we did it inline before because this was inside of a large code block and not a small function. This could also be changed to

```
h.tlsMutex.Lock()
defer h.tlsMutex.Unlock()
h.tlsCert = tlsCert
```

if you think that is clearer (I'm on the fence, since it's just a single line that is in the lock's scope).